doc/man3/OPENSSL_version_text.pod

```diff
@@ -0,0 +1,186 @@
+OPENSSL_version_text, OPENSSL_version_text_full, OPENSSL_info
+- get OpenSSL version number
+
+=head1 SYNOPSIS
+
+ #include <openssl/opensslv.h>
+
+ #define OPENSSL_VERSION_MAJOR  x
+ #define OPENSSL_VERSION_MINOR  y
+ #define OPENSSL_VERSION_PATCH  z
+
+ #define OPENSSL_VERSION_PRE_RELEASE "-dev"
+ #undef OPENSSL_VERSION_BUILD_METADATA
+
+ unsigned int OPENSSL_version_major(void);
+ unsigned int OPENSSL_version_minor(void);
+ unsigned int OPENSSL_version_patch(void);
+ const char *OPENSSL_version_pre_release(void);
+ const char *OPENSSL_version_build_metadata(void);
+
+ const char *OPENSSL_version_text(void);
+ const char *OPENSSL_version_text_full(void);
+
+ #include <openssl/crypto.h>
+
+ const char *OpenSSL_info(int t);
+
+=head1 DESCRIPTION
+
+=head2 Macros
+
+The three macros B<OPENSSL_VERSION_MAJOR>, B<OPENSSL_VERSION_MINOR> and
+B<OPENSSL_VERSION_PATCH> represent the three parts of a 3 numbered version
+number, MAJOR.MINOR.PATCH.
+
+The macro B<OPENSSL_VERSION_PRE_RELEASE> is an added bit of text that,
+when defined, indicates that this is a pre-release version (possibly
+development, for which the value is "-dev").
+
+The macro B<OPENSSL_VERSION_BUILD_METADATA> is extra metadata, reserved
+for other vendors.
+The OpenSSL project will not touch this macro.
+The value must be a string.
+
+=head2 Functions
+
+OPENSSL_version_major(), OPENSSL_version_minor(), OPENSSL_version_patch(),
+OPENSSL_version_pre_release(), and OPENSSL_version_build_metadata() return
+the values of the macros above for the build of the library, respectively.
+
+OPENSSL_version_text() returns the version number "MAJOR.MINOR.PATCH" where
+MINOR, MAJOR and PATCH are the values of B<OPENSSL_VERSION_MAJOR>,
+B<OPENSSL_VERSION_MINOR>, and B<OPENSSL_VERSION_PATCH>, respectively.
+
+OpenSSL_version_text_full() returns the version number text that
+OpenSSL_version_text() returns, with the strings of
+B<OPENSSL_VERSION_PRE_RELEASE> and B<OPENSSL_VERSION_BUILD_METADATA> when
+these are each defined.
+
+OpenSSL_info() returns different strings depending on B<t>:
+
+=over 4
+
+=item OPENSSL_CFLAGS
+
+The compiler flags set for the compilation process in the form
+"compiler: ..."  if available or "compiler: information not available"
+otherwise.
+
+=item OPENSSL_BUILT_ON
+
+The date of the build process in the form "built on: ..." if available
+or "built on: date not available" otherwise.
+
+=item OPENSSL_PLATFORM
+
+The "Configure" target of the library build in the form "platform: ..."
+if available or "platform: information not available" otherwise.
+
+=item OPENSSL_DIR
+
+The "OPENSSLDIR" setting of the library build in the form "OPENSSLDIR: "...""
+if available or "OPENSSLDIR: N/A" otherwise.
+
+=item OPENSSL_ENGINES_DIR
+
+The "ENGINESDIR" setting of the library build in the form "ENGINESDIR: "...""
+if available or "ENGINESDIR: N/A" otherwise.
+
+=back
+
+For an unknown B<t>, the text "not available" is returned.
+
+=head1 BACKWARD COMPATIBILITY
```

Comment:
Please make it clear that they are deprecated

Reply:
Yes